jerry-core/ecma/builtin-objects/ecma-builtin-reflect.c

```diff
@@ -0,0 +1,175 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <math.h>
+#include "ecma-builtins.h"
+#include "ecma-builtin-function-prototype.h"
+#include "ecma-builtin-object.h"
+#include "ecma-exceptions.h"
+#include "ecma-function-object.h"
+#include "ecma-gc.h"
+#include "jcontext.h"
+
+#if ENABLED (JERRY_ES2015_BUILTIN_REFLECT)
+
+#define ECMA_BUILTINS_INTERNAL
+#include "ecma-builtins-internal.h"
+
+/**
+ * This object has a custom dispatch function.
+ */
+#define BUILTIN_CUSTOM_DISPATCH
+
+/**
+ * List of built-in routine identifiers.
+ */
+enum
+{
+  ECMA_REFLECT_OBJECT_ROUTINE_START = ECMA_BUILTIN_ID__COUNT - 1,
+  ECMA_REFLECT_OBJECT_APPLY, /* ECMA-262 v6, 26.1.1 */
+  ECMA_REFLECT_OBJECT_DEFINE_PROPERTY, /* ECMA-262 v6, 26.1.3 */
+  ECMA_REFLECT_OBJECT_GET_OWN_PROPERTY_DESCRIPTOR_UL, /* ECMA-262 v5, 26.1.7 */
+  ECMA_REFLECT_OBJECT_GET_PROTOTYPE_OF_UL, /* ECMA-262 v6, 26.1.8 */
+  ECMA_REFLECT_OBJECT_IS_EXTENSIBLE, /* ECMA-262 v6, 26.1.10 */
+  ECMA_REFLECT_OBJECT_PREVENT_EXTENSIONS_UL, /* ECMA-262 v6, 26.1.12 */
+  ECMA_REFLECT_OBJECT_SET_PROTOTYPE_OF_UL, /* ECMA-262 v6, 26.1.14 */
+};
+
+#define BUILTIN_INC_HEADER_NAME "ecma-builtin-reflect.inc.h"
+#define BUILTIN_UNDERSCORED_ID reflect
+#include "ecma-builtin-internal-routines-template.inc.h"
+
+/** \addtogroup ecma ECMA
+ * @{
+ *
+ * \addtogroup ecmabuiltins
+ * @{
+ *
+ * \addtogroup object ECMA Reflect object built-in
+ * @{
+ */
+
+/**
+ * Dispatcher for the built-in's routines.
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+ecma_value_t
+ecma_builtin_reflect_dispatch_routine (uint16_t builtin_routine_id,
+                                       ecma_value_t this_arg,
+                                       const ecma_value_t arguments_list[],
+                                       ecma_length_t arguments_number)
+{
+  JERRY_UNUSED (this_arg);
+  JERRY_UNUSED (arguments_number);
+
+  if (!ecma_is_value_object (arguments_list[0])
+        && builtin_routine_id != (ECMA_REFLECT_OBJECT_GET_PROTOTYPE_OF_UL
+                                   || ECMA_REFLECT_OBJECT_SET_PROTOTYPE_OF_UL))
```

Comment:
This is not correct.
You want to compare builtin_routine_id to both `ECMA_REFLECT_OBJECT_GET_PROTOTYPE_OF_UL` and `ECMA_REFLECT_OBJECT_SET_PROTOTYPE_OF_UL`. To achieve that you have to correct it like this:
```
if (!ecma_is_value_object (arguments_list[0])
    && (builtin_routine_id != ECMA_REFLECT_OBJECT_GET_PROTOTYPE_OF_UL 
        && builtin_routine_id != ECMA_REFLECT_OBJECT_SET_PROTOTYPE_OF_UL))
```

You could also just modify the enum to have these 2 types as the first 2 members, and check if builtin_routine_id is smaller than the first element that's after them.
This way no long comparison is needed.